Devscord.DiscordFramework/Services/EmbedMessageSplittingService.cs

```diff
@@ -27,20 +30,67 @@ internal IEnumerable<Embed> SplitEmbedMessage(string title, string description,
 
         internal IEnumerable<Embed> SplitEmbedMessage(string title, string description, IEnumerable<KeyValuePair<string, Dictionary<string, string>>> values)
         {
-            var messages = this.SplitMessage(values.ToList()).ToList();
+            var leftForValues = MAX_EMBED_LENGTH - title.Length - description.Length - this._embedMessagesService.FooterLength;
+            var messages = this.SplitMessage(values, leftForValues).ToList();
             yield return this._embedMessagesService.Generate(title, description, messages.FirstOrDefault() ?? new Dictionary<string, Dictionary<string, string>>());
             foreach (var message in messages.Skip(1))
             {
                 yield return this._embedMessagesService.Generate(title: null, description: null, message);
             }
         }
 
-        private IEnumerable<IEnumerable<T>> SplitMessage<T>(IReadOnlyCollection<T> values)
+        private IEnumerable<IEnumerable<KeyValuePair<string, string>>> SplitMessage(IEnumerable<KeyValuePair<string, string>> values, int lengthLeftForValues)
         {
-            for (var i = 0; i < Math.Ceiling((double) values.Count / MAX_FIELDS); i++)
+            var valuesToReturn = new List<KeyValuePair<string, string>>();
+            var valuesToReturnLength = 0;
+            foreach (var value in values)
             {
-                yield return values.Skip(MAX_FIELDS * i).Take(MAX_FIELDS);
+                var valueLength = value.Key.Length + value.Value.Length;
+                var valuesLengthPlusNewValue = valuesToReturnLength + valueLength;
+                if (valuesLengthPlusNewValue > lengthLeftForValues || valuesToReturn.Count == MAX_FIELDS)
+                {
+                    yield return valuesToReturn;
```

Comment:
jakiś dziwny mechanizm jest z tym valuesToReturn
nie dałoby sie zrobić tego "normalnie" po prostu zmieniając troche foreach?

1tworzysz nowy
2 zwracasz go
3 uzupełniasz (pętla)
4 pętla do pkt 2
5 zwracasz go 

nie da sie zrobić żeby było jedno powiedzmy `yield return value`?

Reply:
nie wiem czy dobrze rozumiem o co ci chodzi, ale przez to, że ta metoda zwraca podwójnie zagnieżdżaną kolekcję IEnumerable<IEnumerable<KeyValuePair<string, string>>>to chyba prościej się nie da

Reply:
dobra, to niech zostanie jak jest
do miejsce gdzie nie zagląda sie często + ktoś może chwile sie zastanowić jak to działa